dbms/src/Interpreters/ThreadStatusExt.cpp

```diff
@@ -118,6 +129,74 @@ void ThreadStatus::finalizePerformanceCounters()
     }
 }
 
+namespace {
+    void queryProfilerTimerHandler(int /* sig */, siginfo_t * /* info */, void * context) {
+        DB::WriteBufferFromFileDescriptor out(TracePipe::instance().write_fd);
+
+        std::string queryID = CurrentThread::getCurrentQueryID();
```

Comment:
You cannot create temporary `std::string` object inside signal handler. You cannot even allocate memory.